java/java/src/main/java/org/signal/internal/Native.java

```diff
@@ -72,16 +72,19 @@ private Native() {}
   public static native void Aes256Ctr32_Destroy(long handle);
   public static native long Aes256Ctr32_New(byte[] key, byte[] nonce, int initialCtr);
   public static native byte[] Aes256Ctr32_Process(long ctr, byte[] data);
+  public static native byte[] Aes256Ctr32_ProcessWithOffset(long ctr, byte[] data, int offset, int length);
```

Comment:
Tempted to say the glue layer should *only* have the offset/length APIs and not bother with the whole-array one.

Reply:
Yeah that makes sense. TS might also find that convenient, if/when we ever need to expose these for Desktop.